components/support/nimbus-fml/src/util/loaders.rs

```diff
@@ -407,6 +420,15 @@ impl FileLoader {
     }
 }
 
+impl Drop for FileLoader {
+    fn drop(&mut self) {
+        let cache_dir = self.tmp_cache_dir();
+        if cache_dir.exists() {
+            _ = std::fs::remove_dir_all(cache_dir);
+        }
+    }
+}
```

Comment:
If the `tmp_cache_dir` exists at all it's because:

1. `self.cache_dir` is `None` AND
2. there have been remote FML files been loaded.

There are tiny caveats around this (e.g. that the desktop machine can't derive `SystemClock::now()`, or an exactly named directory in the `temp_dir` is already there — and that's at most a 1 in 1<<48, with someone guessing the correct name for that run).